src/hooks/useStorageValues.jsx

```diff
@@ -0,0 +1,22 @@
+import { useState, useEffect } from 'react';
+
+const useStorageValue = (key) => {
+    const [value, setValue] = useState(localStorage.getItem(key));
+
+    const storageListener = (event) => {
+        console.log(event);
+        if (event.key === key) {
+            setValue(event.newValue);
+        }
+    };
+
+    useEffect(() => {
```

Comment:
are you sure we need this?

Reply:
As I mentioned in one of the previous comments, the component does not subscribe to changes in localstorage. I implemented the subscription in this `useEffect` to resolve this problem.

Reply:
i think it's an overhead

Reply:
Otherwise we have to pass 'navigate' into login thunk to redirect user to home page.